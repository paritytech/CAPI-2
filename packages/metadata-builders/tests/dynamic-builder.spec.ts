import ksm from "./ksm.json"
import { expect, describe, it } from "vitest"
import { getDynamicBuilder } from "@/."

describe("getDynamicBuilder", () => {
  const builder = getDynamicBuilder(ksm as any)
  it("batched call", () => {
    const result = builder
      .buildCall("Utility", "batch")
      .codec.dec(
        "0x08040700dc97b0271418c41f80d049826cfb1d6bd2e44e11ea39759addf6b01632ca973d0b00409452a30306050400dc97b0271418c41f80d049826cfb1d6bd2e44e11ea39759addf6b01632ca973d",
      )

    expect(result).toMatchSnapshot()
  })

  it("felloship referenda submit", () => {
    const result = builder
      .buildCall("FellowshipReferenda", "submit")
      .codec.dec(
        "0x2b0f01590100004901415050524f56455f52464328303030352c39636261626661383035393864323933353833306330396331386530613065346564383232376238633866373434663166346134316438353937626236643434290101000000",
      )
    expect(result).toMatchSnapshot()
  })
})
