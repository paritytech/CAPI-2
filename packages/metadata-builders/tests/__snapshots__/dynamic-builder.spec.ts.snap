// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getDynamicBuilder > batched call 1`] = `
[
  [
    {
      "as": [Function],
      "is": [Function],
      "type": "Balances",
      "value": {
        "as": [Function],
        "is": [Function],
        "type": "transfer",
        "value": {
          "dest": {
            "as": [Function],
            "is": [Function],
            "type": "Id",
            "value": "HZZ7X3nzKuYpdrT7wSDBb8HqB7cc8z77C8oVi2MACzfAhh4",
          },
          "value": 4000000000000n,
        },
      },
    },
    {
      "as": [Function],
      "is": [Function],
      "type": "Staking",
      "value": {
        "as": [Function],
        "is": [Function],
        "type": "nominate",
        "value": {
          "targets": [
            {
              "as": [Function],
              "is": [Function],
              "type": "Id",
              "value": "HZZ7X3nzKuYpdrT7wSDBb8HqB7cc8z77C8oVi2MACzfAhh4",
            },
          ],
        },
      },
    },
  ],
]
`;

exports[`getDynamicBuilder > felloship referenda submit 1`] = `
[
  {
    "as": [Function],
    "is": [Function],
    "type": "Origins",
    "value": {
      "as": [Function],
      "is": [Function],
      "type": "Fellows",
      "value": undefined,
    },
  },
  {
    "as": [Function],
    "is": [Function],
    "type": "Inline",
    "value": {
      "asBytes": [Function],
      "asHex": [Function],
      "asText": [Function],
    },
  },
  {
    "as": [Function],
    "is": [Function],
    "type": "After",
    "value": 1,
  },
]
`;
